### PR TITLE
Add pypi publish on release action

### PR DESCRIPTION
**Pull request by mdegat01**
# Proposed Changes

Add a workflow to publish to pypi on release

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+          with open("pyproject.toml", "wb") as f:
+            tomli_w.dump(pyproject, f)
+      - name: Build python package
+        run: >-
+          python3 -m build
+      - name: Publish release to PyPI
+        uses: pypa/gh-action-pypi-publish@v1.9.0
+        with:
+          user: __token__
+          password: ${{ secrets.PYPI_TOKEN }}
```
Comment (MartinHjelmare) [inline .github/workflows/release.yaml:60]: It's better to configure a trusted publisher at PyPI.

https://packaging.python.org/en/latest/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/

Comment (mdegat01) [reply .github/workflows/release.yaml]: Thanks @MartinHjelmare , I didn't know about that. I was looking at python-matter-server as an example, seems we need to udpate that. I re-wrote using their guide. I also added an option to publish to TestPyPI as a workflow dispatch for testing before releases if we want. Let me know what you think.

Comment (MartinHjelmare) [reply .github/workflows/release.yaml]: Looks great!

**Review by MartinHjelmare — APPROVED** [review]
Nice!

**Pull request merged**